database/rrdcalc.c

```diff
@@ -183,16 +195,24 @@ inline void rrdsetcalc_unlink(RRDCALC *rc) {
 
     rc->rrdset_prev = rc->rrdset_next = NULL;
 
-    rrdvar_free(host, &st->rrdvar_root_index, rc->local);
+    struct rrdcalc_rrdset_alarm *rra = (struct rrdcalc_rrdset_alarm *)avl_search_lock(&host->alarms_idx_health_name, (avl *)search);
+    if(rra) {
+        rrdvar_free(host, &rra->rrdvar_alarm_index, rc->local);
+    }
     rc->local = NULL;
 
-    rrdvar_free(host, &st->rrdfamily->rrdvar_root_index, rc->family);
+    rra = (struct rrdcalc_rrdset_alarm *)avl_search_lock(&host->alarms_idx_health_family, (avl *)search);
+    if(rra) {
+        rrdvar_free(host, &rra->rrdvar_alarm_index, rc->local);
```

Comment:
Shouldn't this be:
```
         rrdvar_free(host, &rra->rrdvar_alarm_index, rc->family); 
```
?

Reply:
You are right, I copied and pasted, but I forgot to change. Thank you!